Wrappers/Python/ccpi/viewer/utils/io.py

```diff
@@ -43,3 +71,761 @@ def SaveRenderToPNG(render_window, filename):
+        if not isinstance(attributes, dict) and not (attributes is None):
+            raise Exception("'attributes' must be a dictionary, or unset (i.e. None)")
+
+    def SetChunking(self, chunking):
+        '''
+        Parameters
+        ----------
+        chunking: bool, default: True in case of HDF5 file,
+                  otherwise False
+            Whether to write to the file in chunks.
+        '''
+        self._Chunking = chunking
+
+    def SetChunkShape(self, chunk_shape):
+        '''
+        Parameters
+        ----------
+        chunk_shape: tuple, default: one slice on the z axis
+            If SetChunking() has been set to True, this will be
+            the size of a chunk written to the file.
+        '''
+        self._ChunkShape = chunk_shape
+
+    def SetHDF5Compression(self, compression):
+        '''
+        Parameters
+        ----------
+        compression: list, default None
+            This is a list of compression settings when writing to HDF5
+            First element: str
+                The type of hdf5 compression to use.
+                Must be one of: None, 'gzip', or 'lzf'
+            Second element:
+                Options for the compression filter. For details, see:
+                https://docs.h5py.org/en/stable/high/dataset.html?highlight=compression_opts#dataset-compression
+            Third element: bool
+                Whether to shuffle the data i.e. rearrange the bytes in
+                a chunk, which may improve the compression ratio
+        '''
+        self._HDF5Compression = compression
+   
+    def Write(self):
+        # check file ext
+        writer = self._GetWriter()
+        writer.Write()
+
+    def _GetWriter(self):
+        file_name = os.path.splitext(self._FileName)[0]
+
+        if self._FileFormat in ['nxs', 'h5', 'hdf5', '']:
+            self._FileName = file_name + '.hdf5'
+            writer = self._GetHDF5Writer()
+
+        elif self._FileFormat in ['mha']:
+            self._FileName = file_name + '.mha'
+            writer = self._GetMetaImageWriter()
+
+        else:
+            raise Exception("File format is not supported. Supported types include hdf5/nexus.")
+
+        writer.SetFileName(self._FileName)
+        return writer
+
+    def _GetHDF5Writer(self):
+        writer = vortexHDF5ImageWriter()
+        writer.SetOriginalDataset(None, self._OriginalDatasetAttributes)
+        writer.SetChunking(self._Chunking)
+        writer.SetChunkShape(self._ChunkShape)
+        writer.SetHDF5Compression(self._HDF5Compression)
+        for i in range(0, len(self._ChildDatasets)):
+            writer.AddChildDataset(self._ChildDatasets[i],  self._ChildDatasetsAttributes[i])
+        return writer
+
+    def _GetMetaImageWriter(self):
+        writer = vtk.vtkMetaImageWriter()
+        writer.SetInputData(self._ChildDatasets[0])
+        return writer
+
+class vortexHDF5ImageWriter(ImageWriter):
```

Comment:
should we name this something else? Are we committed to renaming the module to vortex?

Reply:
rename to cilviewer or clv

Reply:
renamed to cilviewer